File: code/angular/src/app/model3/models/Model3Inputs.ts
export class Model3Inputs {
  startYear: number = 2025;
  ageAtStart: number = 30;
  endYear: number = 2065;
  amountAtStart: number = 100000;
  annualContribution = 12000;
  allocations: Model3Allocation[] = [
    new Model3Allocation('Growth', 0.5, 0.05, 0.25),
    new Model3Allocation('MinimalRisk', 0.5, 0, 0)];
  meanAnnualReturn = 0.05;
  targetAge = 67
}

export class Model3Allocation {
  constructor(public name: string,
    public proportion: number,
    public meanAnnualReturn: number,
    public standardDeviation: number,
  ) {
  }
}

File: code/dotnet/PredictorLibrary/Model3/Inputs/Model3Inputs.cs
namespace PredictorLibrary.Model3.Inputs;

public class Model3Inputs {
    public int StartYear { get; set; }
    public int AgeAtStart { get; set; }
    public int EndYear { get; set; }
    public decimal AmountAtStart { get; set; }
    public decimal AnnualContribution { get; set; }
    public int TargetAge { get; set; }
    
    public IList<Allocation> Allocations { get; set; } = new List<Allocation>();
}